modules/phase_field/src/userobjects/RandomEulerAngleProvider.C

```diff
@@ -0,0 +1,51 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RandomEulerAngleProvider.h"
+#include "MooseRandom.h"
+
+template<>
+InputParameters validParams<RandomEulerAngleProvider>()
+{
+  InputParameters params = validParams<EulerAngleProvider>();
+  params.addClassDescription("Assign random euler angles for each grain.");
+  params.addRequiredParam<UserObjectName>("grain_tracker_object", "The FeatureFloodCount UserObject to get values from.");
+  params.addParam<unsigned int>("seed", 0, "Seed value for the random number generator");
+  return params;
+}
+
+RandomEulerAngleProvider::RandomEulerAngleProvider(const InputParameters & params) :
+    EulerAngleProvider(params),
+    _grain_tracker(getUserObject<GrainTrackerInterface>("grain_tracker_object")),
+    _angles(0)
+{
+  MooseRandom::seed(getParam<unsigned int>("seed"));
```

Comment:
Ugh, :-1:, this is probably my bad as well (since I added the EulerAngles::random method). But this sets the global random seed in the constructor. It can be overwritten by arbitrary other moose objects that use the horrible static MooseRandom interface. Worse: other objects may choose to seed differently on different processors so this object will get inconsistent Euler angles. That should be particularly funky if you have grains crossing the domain decomposition boundaries...